do/build

```diff
@@ -1,26 +1,62 @@
-#!/bin/bash
+#!/usr/bin/env bash
+set -eu -o pipefail
 
-SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-pushd "${SCRIPT_DIR}/.." > /dev/null 2>&1 || exit 1
-# shellcheck source=do/private/script_utils
-source "${SCRIPT_DIR}/private/script_utils"
+function usage()
+{
+  usage_default
+}
+
+function help()
+{
+  help_default
+}
+
+function pre_docker()
+{
+  # this function is called before the docker container runs
+  mkdir -p "${OUTPUT_DIR}"
+
+  if [[ "${BUILD_DOCKER}" -eq 1 ]]; then
+    echo "--- :aws: Log into AWS Elastic Container Registry..."
+    aws_login
```

Comment:
We have decided to leave it because we have work this quarter to mirror Docker Hub images in ECR, so even images like `ubuntu` will be pulled from ECR and require an AWS login.